src/watertap_contrib/seto/unit_models/surrogate/med_tvc_surrogate.py

```diff
@@ -0,0 +1,1394 @@
+from idaes.core.util.scaling import (
+    calculate_scaling_factors,
+    unscaled_variables_generator,
+    badly_scaled_var_generator,
+)
+
+
+@declare_process_block_class("MEDTVCSurrogate")
+class MEDTVCData(UnitModelBlockData):
+    """
+    Multi-effect distillation with thermal vapor compressor model
+    """
+
+    CONFIG = ConfigBlock()
+
+    CONFIG.declare(
+        "dynamic",
+        ConfigValue(
+            domain=In([False]),
+            default=False,
+            description="Dynamic model flag - must be False",
+            doc="""Indicates whether this model will be dynamic or not,
+    **default** = False. """,
+        ),
+    )
+    CONFIG.declare(
+        "has_holdup",
+        ConfigValue(
+            default=False,
+            domain=In([False]),
+            description="Holdup construction flag - must be False",
+            doc="""Indicates whether holdup terms should be constructed or not.
+    **default** - False.""",
+        ),
+    )
+    CONFIG.declare(
+        "property_package_water",
+        ConfigValue(
+            default=useDefault,
+            domain=is_physical_parameter_block,
+            description="Property package to use for feed, distillate, brine and cooling water properties",
+            doc="""Property parameter object used to define water property calculations,
+    **default** - useDefault.
+    **Valid values:** {
+    **useDefault** - use default package from parent model or flowsheet,
+    **PhysicalParameterObject** - a PhysicalParameterBlock object.}""",
+        ),
+    )
+    CONFIG.declare(
+        "property_package_steam",
+        ConfigValue(
+            default=useDefault,
+            domain=is_physical_parameter_block,
+            description="Property package to use for heating and motive steam properties",
+            doc="""Property parameter object used to define steasm property calculations,
+    **default** - useDefault.
+    **Valid values:** {
+    **useDefault** - use default package from parent model or flowsheet,
+    **PhysicalParameterObject** - a PhysicalParameterBlock object.}""",
+        ),
+    )
+    CONFIG.declare(
+        "property_package_args",
+        ConfigBlock(
+            implicit=True,
+            description="Arguments to use for constructing property packages",
+            doc="""A ConfigBlock with arguments to be passed to a property block(s)
+    and used when constructing these,
+    **default** - None.
+    **Valid values:** {
+    see property package for documentation.}""",
+        ),
+    )
+
+    def build(self):
+        super().build()
+
+        self.scaling_factor = Suffix(direction=Suffix.EXPORT)
+
+        units_meta = self.config.property_package_water.get_metadata().get_derived_units
+
+        """
+        Add system configurations
+        """
+        self.number_effects = Param(
+            initialize=12,
+            mutable=True,
+            within=Set(initialize=[8, 10, 12, 14, 16]),
```

Comment:
Are these the only values that are considered validated by PSA? That is, should the user only choose from these values explicitly? Can you add a comment above this explaining the Set choice and how it applied to limitations of the model?

Reply:
My response was:
The range is wider but these numbers are the ones that were validated. For value like 9,11, and etc., we can do linear interpolation. Do we want to implement this feature?

Actually I'm thinking maybe the number of effect should be a config block itself. Does it make sense?

Reply:
Yes, that makes sense, and I think you could go ahead and make it a config option if you like. Also, I think we can at least include the intermediate values for interpolation.

Reply:
@Zhuoran29 do you want to implement as CONFIG option? I am ok with it either way. This is the way we did it for LT-MED though. But if you do add it as a CONFIG option, make sure to add a test for it.

But also, it the model works well (i.e. doesn't give insane numbers) for number effects not in the Set, that might be useful to include.